### DISCUSSION TITLE
Significant latency when multiple users use the shared filesystem

### DISCUSSION THREAD

**Discussion by ahmedTouati** [rook/rook#8621] [Q&A]
We have a cluster-on-pvc installation in our Kubernetes cluster with 3 OSDs each runs on a t3.xlarge (4CPU, 16GB RAM, 5Gbps Bandwith). However, when multiple clients connect to the cephFS (e.g executing **mvn install**), most of these operations takes a long time to complete, we still cannot find the root cause of this latency, the only thing that's getting fired all the time is the **MDS cache oversized** and **Clients failing to respond to cache pressure** warnings, I know that this has to do with the **mds_cache_memory_limit** parameter but does increasing it can solve the problem? Another thing we constated in our grafana dashboard is that the average network bandwidth used by ceph cluster is only 250Mbps while our nodes have 5Gbps, which means that our cluster is probably not fully utilizing the available bandwidth, which we think is another reason behind that latency.

**Comment by BlaineEXE** [comment]
It sounds like you have 3 nodes, each with 1 OSD. I suspect that bandwidth to each OSD is about 250Mbps, that being your bottleneck. Ceph performance increases with more disks per node. If you want to make the most of your 5Gbps network, you can probably have approximately 20 OSDs per node easily. Likely a bit more because Ceph usually won't be writing to all disks at once except when it is recovering from a node failure. I suspect that if you add OSDs, you will find performance improves, including latency.

**Reply by ahmedTouati** [reply]
@BlaineEXE What about the MDS cache memory, if we increase its limit, wouldn't that help ?

**Reply by BlaineEXE** [reply]
Possibly, but that will only help for file metadata (things like inodes) to my understanding. You will still have the bottleneck that I mentioned for manipulating content of files.
